6115: add note about onactivateasync to transactions part

RamanBut-Husaim opened the pull request:
A documentation has been extended with the note related to transactions & `onactivateasync` based on the discussion inside of https://github.com/dotnet/orleans/issues/6115.

RamanBut-Husaim commented:
@jason-bragg Please take a look.

RamanBut-Husaim commented:
@jason-bragg Hello. Sorry for pinging you with this. Maybe there is something to change as part of this PR?
Thanks.

jason-bragg commented:
@RamanBut-Husaim, I apologize for the delayed follow-up, and thank you for the contribution.  My only issue with the comment is the "at the moment" part.  That seems to imply that we've plans to change this behavior, which we don't.  It would be better if we could access transactional state on activation, but the current patterns don't support this and addressing this is not yet on any of our road maps.  It's best, imo, not to give an impression that it is.

RamanBut-Husaim commented:
@jason-bragg Hey. 

Thanks a lot for the feedback. Ok, understood. I've removed `at the moment` part and rebased on the latest `docs` branch.

Could you please take a look when you have time?

Thanks.

sergeybykov commented:
Thank you, @RamanBut-Husaim! Sorry about delays.

RamanBut-Husaim commented:
@sergeybykov @jason-bragg Thanks a lot!

That is fine, Sergey. I perfectly understand that you are extremely busy with other stuff.
